packages/flutter/lib/src/widgets/page_view.dart

```diff
@@ -848,10 +842,12 @@ class PageView extends StatefulWidget {
 class _PageViewState extends State<PageView> {
   int _lastReportedPage = 0;
 
+  late final PageController _controller = widget.controller ?? PageController();
```

Comment:
I would suspect that this will break if you first build a PageView with a controller and then rebuild it without the controller. The state would still operate on the no-longer-provided controller, which may have been disposed already. You'll probably have to deal with this situation in didUpdateWidget.

(We should also add a test for this scenario if we don't already have one)